### PR TITLE
Fixes issue #5246 Add Reference quick fix does not add a reference The

### PR DESCRIPTION
**Pull request by mkrueger**
AddMissingReferenceCodeAction uses a specialized service in the case
of system assemblies.
For other cases file & project references the workspace is queried
directly.

### CODE REVIEW

**Comment by mkrueger** [comment]
@monojenkins backport release-7.6

**Comment by p-lad** [comment]
Hello,
Tested this pr on the build mentioned in the commits section and it is working fine.Reference was added to ExportAttributeTest mentioned in bug #641610

Build taken:
https://github.com/mono/monodevelop/pull/5261/commits

Build info:
Version 7.7 Preview (7.7 build 407)
 
Build gist:
https://gist.github.com/p-lad/a3ae6c55b8d158c09f2325f6393cfddc
 
Screencast link:
https://www.screencast.com/t/63OnjxsCZZhs

Thanks.

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.TypeSystem/MonoDevelopAddMetadataReferenceCodeActionOperationFactoryWorkspaceService.cs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+			public AddMetadataReferenceOperation (ProjectId projectId, AssemblyIdentity assemblyIdentity)
+			{
+				this.projectId = projectId;
+				this.assemblyIdentity = assemblyIdentity;
+			}
+
+			public override void Apply (Workspace workspace, CancellationToken cancellationToken = default (CancellationToken))
+			{
+				var monoDevelopProject = (MonoDevelopWorkspace)workspace;
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.TypeSystem/MonoDevelopAddMetadataReferenceCodeActionOperationFactoryWorkspaceService.cs]: We should do a weak cast here and not do anything if it's not a MonoDevelopWorkspace (see razor)

Comment (mkrueger) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.TypeSystem/MonoDevelopAddMetadataReferenceCodeActionOperationFactoryWorkspaceService.cs]: The VS.NET implementation of this service doesn't do a weak cast. In which scenarios a code action is issued and it's not a monodevelop workspace ?

**Comment by slluis** [comment]
@mkrueger can you fix the conflict and backport the last commit to 7.6?

**Comment by mkrueger** [comment]
@monojenkins backport release-7.6

**Pull request merged**